src/delete.rs

```diff
@@ -183,7 +183,7 @@ mod tests {
         fn all(tests: impl IntoIterator<Item = Test<V>>) {
             tests.into_iter().enumerate().for_each(|(i, t)| t.run(i));
         }
-        fn run(self, i: usize) {
+        fn run(self, _i: usize) {
```

Comment:
So initially the `i` was for the index of the test itself. It was primarily added so that I could easily reconcile which test was failing. I left it in the event we run into failing tests in the future.

I guess the need for `i` kind of demonstrates the short-comings of table-based tests. Had they all been independent tests, this would have been a non-issue.

Reply:
I forgot to mention why it became `_i`. The usage of `i` was used in the output of the `assert` calls. They were pulling down the code coverage and I figured if they are all passing, no need for the message... sigh, my silly ass attempt to reach 100.

Reply:
The `i` is fine, although perhaps having descriptive labels would be best. Not something to change in this PR, but worth thinking about if you make it a trait in the future.